Update Docker images and related documentation (#535)
chaoss/grimoirelab: docker/README.md

# Docker image for GrimoireLab

## Intro

This docker image runs a container preinstalled with a certain release of

> GrimoireLab, including all libraries and programs. By default, it runs 
> `sirmordred` for producing a dashboard given a mordred configuration file.

Reviewer: ```suggestion
GrimoireLab, including all libraries and programs. By default, it runs 
the GrimoireLab orchestrator - `sirmordred` - for producing a dashboard given a mordred configuration file.
```

> If you prefer, you can run SirMordred using docker-compose it's the simplest 
> method to get started with. More details in the [docker-compose 
> folder](../docker-compose/).

Reviewer: ```suggestion
If you prefer, you can run GrimoireLab using docker-compose it's the simplest 
method to get started with. More details in the [docker-compose 
folder](../docker-compose/).
```

## Requirements

* [Git](https://git-scm.com/)
* [Docker](https://docs.docker.com/get-docker/) and [Docker Compose](https://docs.docker.com/compose/install/)
* Recommended, 2CPUs, 8GB memory RAM, and 2GB SWAP (MacOS users can manage it 
with [Docker client for Mac](https://hub.docker.com/editions/community/docker-ce-desktop-mac))

## Build the image

To produce the image, run the following command:
```
$ docker build -f Dockerfile -t grimoirelab/grimoirelab .
```

You can build a specific version of grimoirelab updating `GRIMOIRELAB_RELEASE`
environment variable inside the Dockerfile.

## Run the image

You can run any GrimoireLab tool using this image, for example, you can collect 
Git commits using Perceval:
```

> $ docker run grimoirelab/grimoirelab perceval git https://github.com/chaoss/grimoirelab-perceval.git

Reviewer: This should go after the next paragraph because it's not the default use. I would rephrase it as `You can also run ...`

```

> By default, this image will run `sirmordred`. You need to define a

Reviewer: ```suggestion
The image will run the GrimoireLab orchestrator -  `sirmordred`- by default. You need to define a
```

directory. These files are the minimum set up to analyze git activity in this 
repository. If you want to analyze a different project with a different set of 
repositories and data sources, you need to modify [`projects.json`](../default-grimoirelab-settings/projects.json) 
and [`setup.cfg`](../default-grimoirelab-settings/setup.cfg) files. See 
[below](#more-information) more information about these files format.

Default configuration files assumes Elasticsearch is available in port 9200, 
and MariaDB or MySQL in their standard port (if they are not present, the 
process will fail).

Run the following command from the root of this repository to start using 
SirMordred:
```
docker run --net=host \ 
    -v $(pwd)/default-grimoirelab-settings/projects.json:/home/bitergia/conf/projects.json \
    -v $(pwd)/default-grimoirelab-settings/setup-docker.cfg:/home/bitergia/conf/setup.cfg \
    -v $(pwd)/default-grimoirelab-settings/organizations.json:/home/bitergia/organizations.json \
    -v $(pwd)/default-grimoirelab-settings/identities.yml:/home/bitergia/conf/identities.yml \
    -t grimoirelab/grimoirelab
```

> Remember you can run SirMordred and this image using docker-compose. It's the 

Reviewer: ```suggestion
Remember you can run GrimoireLab and this image using docker-compose. It's the 
```

simplest method to get started with. More details in the [docker-compose 
folder](../docker-compose/).

> ## SirMordred configuration

Reviewer: ```suggestion
## Configuration
```

The different files we can modify are:

> - setup.cfg: SirMordred's configuration file

Reviewer: ```suggestion
- setup.cfg: SirMordred's configuration file (GrimoireLab orchestrator configuration)
```

- projects.json: JSON file with projects and their repositories

### setup.cfg

> SirMordred's configuration file is composed of several sections which allow to

Reviewer: ```suggestion
The configuration file is composed of several sections which allow to
```
